common/lib/capa/capa/xqueue_interface.py

```diff
@@ -134,12 +136,18 @@ def _send_to_queue(self, header, body, files_to_upload):
 
     def _http_post(self, url, data, files=None):
         try:
-            r = self.session.post(url, data=data, files=files)
+            response = self.session.post(
+                url, data=data, files=files, timeout=(CONNECT_TIMEOUT, READ_TIMEOUT)
+            )
         except requests.exceptions.ConnectionError, err:
             log.error(err)
             return (1, 'cannot connect to server')
 
-        if r.status_code not in [200]:
-            return (1, 'unexpected HTTP status code [%d]' % r.status_code)
+        except requests.exceptions.ReadTimeout, err:
```

Comment:
Please try to combine two `except` blocks (if possible) . Something like 
`except request.exceptions, err:` 
`log.error(err)` 
`error_message = 'cannot  connect to server'`
`if (err. ReadTimeout):` 
`error_message = 'failed to read from the server'`
`return (1, error_message)`